src/core/net/checksum.hpp

```diff
@@ -84,6 +85,30 @@ class Checksum
                                       const Ip6::Address &aDestination,
                                       uint8_t             aIpProto);
 
+    /**
+     * This static method calculates and then updates the checksum in a given IPv4 message (if TCP/UDP/ICMP(v4)).
+     *
+     * @param[in,out] aMessage  The message to update the checksum in. The `aMessage.GetOffset()` should point to start
+     *                          of the TCP/UDP/ICMP(v4) header. On exit the checksum field in TCP/UDP/ICMP(v4) header in
+     *                          the message is updated.
+     * @param[in] aSource       The source address.
+     * @param[in] aDestination  The destination address.
+     * @param[in] aIpProto      The Internet Protocol value.
+     *
+     */
+    static void UpdateMessageChecksum(Message &           aMessage,
+                                      const Ip4::Address &aSource,
+                                      const Ip4::Address &aDestination,
+                                      uint8_t             aIpProto);
+
+    /**
+     * This static method calculates and then updates the checksum field in the IPv4 header.
+     *
+     * @param[in,out] aHeader The IPv4 header to update the checksum in.
+     *
+     */
+    static void UpdateIPv4HeaderChecksum(Ip4::Header &aHeader);
```

Comment:
nit: we typically use `Ip6` in code. How about using `Ip4` to be consistent?
```suggestion
    static void UpdateIp4HeaderChecksum(Ip4::Header &aHeader);
```